Tests/SystemTests/FilePathTests/FilePathComponentsTest.swift

```diff
@@ -244,11 +244,15 @@ final class FilePathComponentsTest: XCTestCase {
       TestPathComponents("/foo///bar", root: "/", ["foo", "bar"]),
       TestPathComponents("foo/bar/", root: nil, ["foo", "bar"]),
       TestPathComponents("foo///bar/baz/", root: nil, ["foo", "bar", "baz"]),
-      TestPathComponents("//foo///bar/baz/", root: "/", ["foo", "bar", "baz"]),
       TestPathComponents("./", root: nil, ["."]),
       TestPathComponents("./..", root: nil, [".", ".."]),
       TestPathComponents("/./..//", root: "/", [".", ".."]),
     ]
+#if !os(Windows)
+    testPaths += [
+      TestPathComponents("//foo///bar/baz/", root: "/", ["foo", "bar", "baz"]),
+    ]
```

Comment:
Hmm, I need to check, but that is what I would expect.  However, I don't remember if this was a case where we were asserting incorrectly or not.

Reply:
I believe we fix up incomplete roots and make them complete. C# seems to be pretty inconsistent / emergent-behaviorish here, but that's the only sensible model I could think of at the time.

Reply:
I think that the model is correct in that the behaviour is what we would like.  The question is more about whether the behaviour matches the implementation or not (aka, there is a bug - I think that we were asserting on an empty path component somewhere?).  Sorry if it came across as I didn't think that the behaviour was correct.

Reply:
Path components cannot be empty, but I wonder if we should allow empty server or share portions of a UNC path type. Sorry, I've paged a lot of this out and am just getting back up to speed.

Reply:
```pwsh
> [IO.Path]::GetFullPath("//foo///bar/baz/")
\\foo\bar\baz\
> [IO.Path]::GetPathRoot("//foo///bar/baz/")
\\foo\bar
> [IO.Path]::GetPathRoot([IO.Path]::GetFullPath("//foo///bar/baz/"))
\\foo\bar
```

This also matches the Microsoft documentation on path normalization:
> Normally, any path passed to a Windows API is (effectively) passed to the [GetFullPathName function](https://learn.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-getfullpathnamea) and normalized. There is one important exception: a device path that begins with a question mark instead of a period. Unless the path starts exactly with \\?\ (note the use of the canonical backslash), it is normalized.